### ISSUE TITLE
module exports do not filter values during type checking

### ISSUE THREAD

**Issue by seliopou** [elm/compiler#278]
Say you write a module called `Else`:

``` haskell
module Else ( theNumberOne ) where

theNumberOne = 1
theNumberTwo = 2
```

... that you then import in a module called `Something` and use both names defined by `Else`, even though only `theNumberOne` is exported by it:

``` haskell
module Something where

import open Else

theNumberThree = theNumberOne + theNumberTwo

main = asText theNumberThree
```

Even though `theNumberTwo` is not exported by `Else`, this does not produce a compiler error. But when you open the generated HTML file, it will display the text `NaN`.
